src/components/FormContext.tsx

```diff
@@ -31,11 +31,29 @@ export const FormContext: FC<FormContextProps> = props => {
     submitFocusError: submitFocusError,
   })
 
+  const idMap = useMemo(() => getIdSchemaPairs(props.schema), [props.schema])
+
+  const resolvedSchemaRefs = useMemo(
+    () => resolveRefs(props.schema, idMap, []),
+    [props.schema, idMap]
+  )
+
+  const formContext: JSONFormContextValues = {
+    ...methods,
+    schema: resolvedSchemaRefs,
+    idMap: idMap,
+    customValidators: props.customValidators,
+  }
+
   const formProps: React.ComponentProps<'form'> = {}
 
-  formProps.onSubmit = methods.handleSubmit((data, event) => {
+  formProps.onSubmit = methods.handleSubmit(async (data, event) => {
     if (props.onSubmit) {
-      props.onSubmit(getObjectFromForm(props.schema, data), event)
+      return await props.onSubmit({
```

Comment:
Maybe this await could be removed...
https://github.com/vtex/javascript/blob/2682cbba05c24447f586fa0360fb757241b83342/packages/eslint-config-vtex/rules/best-practices.js#L152